### DISCUSSION TITLE
Heatit TF016 / TF021 - Basic report

### DISCUSSION THREAD

**Discussion by kjetilsn** [zwave-js/zwave-js#6946] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

This relates to older Heatit thermostats TF016 / TF021 which does not report power usage or relay state. In order to calculate power usage we have used the On/Off control group association back to the gateway to get the state. I will admit that I don't know if the "bug" (i guess it is not really a bug) is in Z-wave JS, or in home assistant.
In the past a number entity has been available in home assistant for basic report from the zwave device for knowing the state of the relay. This worked when the on/off association were established between the node and gateway. A while ago this was replaced by a light entity, I suspect this was not intentional, but it worked. However, now there are no entities available for this information, nor is it available as i.e zwave_js.value_updated in the events listener I home assistant. Z-wave JS logs the information, so the node does send it:

``
2024-06-14T07:22:23.727Z CNTRLR   [Node 005] [~] [Basic] currentValue: 99 => 99                     [Endpoint 0] 2024-06-14T07:22:23.733Z DRIVER « [Node 005] [REQ] [ApplicationCommand] └─[BasicCCReport] current value: 99 2024-06-14T07:22:23.734Z CNTRLR   [Node 005] [~] [Basic] currentValue: 99 => 99 
``

### Software versions

Driver (node-zwave-js): 12.10.0

Z-Wave JS UI:  9.13.4.e803879

### Device information

Manufacturer: Heatit
Model name: TF016 / TF021
Node ID: 006

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [ ] The log includes the problematic interaction with the device _(if applicable)_.

- [x] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwave_js.log](https://github.com/user-attachments/files/15871656/zwave_js.log)
[zwave_js-a52298b972a36aa7260d504c814976b8-Bad-b2e9f709f4337079d6f4bd0988ee018c.json](https://github.com/user-attachments/files/15871672/zwave_js-a52298b972a36aa7260d504c814976b8-Bad-b2e9f709f4337079d6f4bd0988ee018c.json)

**Comment by AlCalzone** [comment]
Have you tried re-interviewing the device?

**Reply by kjetilsn** [reply]
I have, no change.
I think it is related to this:
https://github.com/zwave-js/node-zwave-js/pull/6485

**Reply by AlCalzone** [reply]
Please share a log of the re-interview, so I can try to reproduce this.

**Comment by kjetilsn** [comment]
Thank you for looking into this!

I had a look at the device json file and perhaps found the cause, the value mapBasicReport": false?
Attached is json file and debug log during re-interview.


`    "state": {
      "nodeId": 6,
      "index": 0,
      "installerIcon": 4608,
      "userIcon": 4608,
      "status": 4,
      "ready": true,
      "isListening": true,
      "isRouting": true,
      "isSecure": false,
      "manufacturerId": 411,
      "productId": 1,
      "productType": 1,
      "firmwareVersion": "1.8",
      "zwavePlusVersion": 1,
      "name": "Bad",
      "location": "**REDACTED**",
      "deviceConfig": {
        "filename": "/usr/src/app/store/.config-db/devices/0x019b/tf016_tf021.json",
        "isEmbedded": true,
        "manufacturer": "Heatit",
        "manufacturerId": 411,
        "label": "TF016 / TF021",
        "description": "Z-Wave Thermostat",
        "devices": [
          {
            "productType": 1,
            "productId": 1
          },
          {
            "productType": 3,
            "productId": 513
          }
        ],
        "firmwareVersion": {
          "min": "0.0",
          "max": "255.255"
        },
        "preferred": false,
        "associations": {},
        "paramInformation": {
          "_map": {}
        },
        "compat": {
          "mapBasicReport": false`




[re_interview.txt](https://github.com/user-attachments/files/15953323/re_interview.txt)
[zwave_js-a52298b972a36aa7260d504c814976b8-Bad-b2e9f709f4337079d6f4bd0988ee018c (2).json](https://github.com/user-attachments/files/15953324/zwave_js-a52298b972a36aa7260d504c814976b8-Bad-b2e9f709f4337079d6f4bd0988ee018c.2.json)

**Reply by AlCalzone** [reply]
Did you have some filters enabled? If so, please disable them and try again. I don't see the outgoing commands in that log.

`"mapBasicReport": false` means "treat Basic Report as-is, don't try to be smart": https://zwave-js.github.io/node-zwave-js/#/config-files/file-format?id=mapbasicreport

**Reply by kjetilsn** [reply]
Ok, that makes sense.
I might have cut the log short, here is another try:
[re_interview_c.txt](https://github.com/user-attachments/files/15954615/re_interview_c.txt)

**Reply by AlCalzone** [reply]
Can you try using Z-Wave JS UI for the logging?
I see only half the log contents - no outgoing commands, no incoming commands. Only the responses from the stick itself.
Something is filtering or dropping information from the log.

**Reply by kjetilsn** [reply]
Please see z-wave js us log file below:

[zwavejs_2024-06-24.log](https://github.com/user-attachments/files/15955351/zwavejs_2024-06-24.log)

**Reply by AlCalzone** [reply]
Same thing (also there's no interview in that log). Please check what you have here in the settings. It should be empty.
![grafik](https://github.com/zwave-js/node-zwave-js/assets/17641229/54c25344-4656-4bbd-a204-8190f49e09aa)

**Reply by kjetilsn** [reply]
That is strange, I have the same setting, no spesific node selected:
![image](https://github.com/zwave-js/node-zwave-js/assets/35525534/784cc3f2-0735-436a-b219-1ec60d97c564)

**Reply by AlCalzone** [reply]
You're looking at the wrong section (General instead of Z-Wave)

**Comment by kjetilsn** [comment]
Changed from "debug" to "silly":
[zwavejs_2024-06-24_b.log](https://github.com/user-attachments/files/15955741/zwavejs_2024-06-24_b.log)

**Reply by kjetilsn** [reply]
This is strange, after repeatadly re-interview and turning on and off logging I now have the basic light entity!
I tried to replicate on another device, first re-interview in HA, then in Zwave-js UI, restarted HA each time to no success, however when turning on logging in Zwave JS UI (which seem to restart Zwave-JS), then re-interview seems to solve it. Sorry for wasting your time, I guess there are no errors here

**Comment by kjetilsn** [comment]
Aha,
I'm doing most things wrong today it seems. Yes there were a filter for a single node..
I now slowly get one by one working by repeatably restarting zwave-js and doing re-interview just after restart, weirdly this works. 
[zwavejs_2024-06-24_d.log](https://github.com/user-attachments/files/15956216/zwavejs_2024-06-24_d.log)

**Reply by AlCalzone** [reply]
Thanks - I'll try to reproduce anyways, since there's still something that bothers me.

**Reply by kjetilsn** [reply]
Ok, thank you very much!

**Reply by AlCalzone** [reply, marked as answer]
escalated to an issue: https://github.com/zwave-js/node-zwave-js/issues/6963

**Reply by nesheimt** [reply]
Hi @kjetilsn 
Seems like you made your Heatit TF016/TF021 work after this fix. 

I have a number of these thermostats in Homseer4 that is using the zwave-js-ui: 9.20.0 / zwave-js: 13.3.1 but the temperatures are not reporting back to the controller, nor the relay status. Associations are not able to add or remove. 

I'm trying to operate the thermostat from the zwave js ui frontend so that any "bad" impact from the controller should be negligible.

Does the Heatit TF016/TF021 still work fine from your system?

**Reply by AlCalzone** [reply]
@nesheimt open a new discussion with your information, then I can investigate if it's the same issue or something else.

**Reply by nesheimt** [reply]
Hi @AlCalzone 
I have now to the best of my ability created a discussion and entered the requied information and attached logfiles and dumps.
https://github.com/zwave-js/node-zwave-js/discussions/7262

**Comment by kjetilsn** [comment]
Hello again,
I updated to latest docker containing v12.12.1 and updated latest HA docker as well, hoping that would be the solution.
Still the problem remains. Actually the few nodes where I got the basic light entity back are now gone again, and don't come back when doing interview and restart. attached a fresh log of interview.

[re_interview_e.txt](https://github.com/user-attachments/files/16010355/re_interview_e.txt)

**Reply by AlCalzone** [reply]
Can you double check what you see in Z-Wave JS UI? There's also a new option to export a node dump from the driver, this would also be helpful.

**Comment by kjetilsn** [comment]
Thank you.
What specifically should I check in Z-Wave JS UI? The Basic v0 values does update. 
Attached is the dump of the node I previously shared the interview log for.
[node_9_dump.json](https://github.com/user-attachments/files/16010594/node_9_dump.json)

**Reply by AlCalzone** [reply]
Ok then I think you'll just have to wait a bit and update HA. These values will be exposed as sensors, not light entities in 2024.7.0b0:
https://github.com/home-assistant/core/pull/105134

**Reply by kjetilsn** [reply]
Thank you,
I will wait for next release and test then.

**Reply by kjetilsn** [reply]
Hello again,

Just wanted to report that my issue is now solved with the new basic sensor entities with the latest version of home assistant, thank you very much for fixing this!
